### DISCUSSION TITLE
Failed to AOT compile Microsoft.Maui.Controls.dll

### DISCUSSION THREAD

**Discussion by williambuchanan2** [dotnet/maui#11432] [General]
I just updated my Maui app to .NET 7 and I now get this error when building for ios in Azure Devops Pipeline.

Any thoughts on how to fix this?

Thanks in advance...

**Comment by FM1973** [comment]
I had the same problem.
I am using SKIA-Sharp and got the same error not pointing that direction.
I had to add the following to the project file:

`<PropertyGroup>
    <AndroidLinkResources>false</AndroidLinkResources>
  </PropertyGroup>`

**Reply by williambuchanan2** [reply]
Thanks for the suggestion. Unfortunately didn't work though...

**Comment by NikolaJackova** [comment]
I had the same issue with building MAUI App. Try to include this code into csproj:
`<PropertyGroup>
    <UseInterpreter>true</UseInterpreter>
</PropertyGroup>`

**Comment by williambuchanan2** [comment]
It turned out to be an issue with the runtimeidentifier and codesigning. I had to put the following in the csproj:

	<PropertyGroup Condition="$(TargetFramework.Contains('-ios')) and '$(Configuration)' == 'Release'">
		<RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
	</PropertyGroup>

But also had to play around with the provisioning profile and codesign key to get it working.

The trouble is the error message is very misleading. I believe it could be caused by a lot of things.
